internal/service/project/resource_project_test.go

```diff
@@ -1080,6 +1080,74 @@ func TestAccProject_withTags(t *testing.T) {
 	})
 }
 
+func TestAccProject_slowOperationNotOwner(t *testing.T) {
+	var (
+		orgID                    = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName              = acc.RandomProjectName()
+		config                   = configBasic(orgID, projectName, "", false, nil, conversion.Pointer(false))
+		providerConfigPublicKey2 = acc.ConfigPublicKey2Provider()
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t); acc.PreCheckPublicKey2(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             acc.CheckDestroyProject,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "is_slow_operation_thresholding_enabled", "false"),
+				),
+			},
+			{
+				PreConfig: func() { changeRoles(t, orgID, projectName, "GROUP_READ_ONLY") },
+				Config:    providerConfigPublicKey2 + config,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "is_slow_operation_thresholding_enabled", "false"),
+				),
+			},
+			// Validate the API Key has a different role
+			{
+				Config:      providerConfigPublicKey2 + configBasic(orgID, projectName, "", false, nil, conversion.Pointer(true)),
+				ExpectError: regexp.MustCompile("error in project settings update"),
```

Comment:
Originally, tried to modify existing API key role, but didn't getting an error on update.
Therefore, I had to add another API Key